planner/select.go

```diff
@@ -296,20 +296,22 @@ func (n *selectNode) initSource() ([]aggregateNode, error) {
 	return aggregates, nil
 }
 
-func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.FieldDescription] {
+func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.IndexDescription] {
 	if scanNode.filter != nil {
-		schema := scanNode.col.Schema()
-		indexedFields := scanNode.col.Description().CollectIndexedFields(&schema)
-		for i := range indexedFields {
-			typeIndex := scanNode.documentMapping.FirstIndexOfName(indexedFields[i].Name)
-			if scanNode.filter.HasIndex(typeIndex) {
-				// we return the first found indexed field to keep it simple for now
-				// more sophisticated optimization logic can be added later
-				return immutable.Some(indexedFields[i])
+		colDesc := scanNode.col.Description()
+
+		for _, field := range scanNode.col.Schema().Fields {
+			if _, isFiltered := scanNode.filter.ExternalConditions[field.Name]; !isFiltered {
+				continue
+			}
+			indexes := colDesc.CollectIndexesOnField(field.Name)
+			if len(indexes) > 0 {
+				return immutable.Some(indexes[0])
```

Comment:
question:  This looks suspicious, why are you just returning the first?  What if there are multiple indexes using the same field? It that tested?

Reply:
it will be part of another PR. 

Added a comment.

Reply:
Okay cheers.  Can you please make sure that there are integration tests that make sure that if there are multiple overlapping indexes (e.g. two composites, a simple and a composite) that queries still work (no need for explain test documenting this issue, but that would be nice if you can do it reliably without much effort too)?

Reply:
This line of code is not new. There are integration tests that cover it